core/main/src/processor/metrics_processor.rs

```diff
@@ -214,7 +214,10 @@ impl ExtnRequestProcessor for MetricsProcessor {
                 }
             }
             MetricsPayload::TelemetryPayload(t) => {
-                TelemetryBuilder::update_session_id_and_send_telemetry(&state, t).is_ok()
+                let result =
+                    TelemetryBuilder::update_session_id_and_send_telemetry(&state, t).is_ok();
+
+                result & Self::ack(client, msg).await.is_ok()
```

Comment:
can this just be: 

TelemetryBuilder::update_session_id_and_send_telemetry(&state, t).is_ok() && Self::ack(client, msg).await.is_ok()
?

Reply:
Got rid of `result` -- thanks!